apps/native/app/(tabs)/wallet/[walletId].tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useMemo, useState } from "react";
+import {
+    SafeAreaView,
+    ScrollView,
+    StyleSheet,
+    Text,
+    TouchableOpacity,
+    View,
+} from "react-native";
+import { Stack, useLocalSearchParams } from "expo-router";
+import { WalletActivity } from "../../../components/wallet/WalletActivity";
+import { WalletDetails } from "../../../components/wallet/WalletDetails";
+import { Colors } from "../../../constants/Colors";
+import { getPlaceholderWalletActivity } from "../../../data/walletActivityPlaceholder";
+
+const mapParamsToString = (param: string | string[] | undefined) =>
+    Array.isArray(param) ? param.join("") : param ?? "";
+
+export default function WalletDetail() {
+    const params = useLocalSearchParams();
+    const walletId = mapParamsToString(params.walletId);
+    const walletName = mapParamsToString(params.name) || "Wallet";
+    const balanceValue = Number.parseFloat(mapParamsToString(params.balance));
+    const walletBalance = Number.isFinite(balanceValue) ? balanceValue : 0;
+    const [activeTab, setActiveTab] = useState<"activity" | "details">(
+        "activity",
+    );
+
+    const { pending, completed, completedLabel } = useMemo(
+        () => getPlaceholderWalletActivity(walletId),
+        [walletId],
+    );
+
+    return (
+        <>
+            <Stack.Screen
+                options={{
+                    title: walletName,
+                    headerShown: true,
+                    headerShadowVisible: false,
+                }}
+            />
+            <SafeAreaView style={styles.safeArea}>
+                <ScrollView
+                    style={styles.scrollView}
+                    contentContainerStyle={styles.scrollContent}
+                    showsVerticalScrollIndicator={false}>
+                    <View style={styles.tabBar}>
+                        <TabButton
+                            label="Activity"
+                            isActive={activeTab === "activity"}
+                            onPress={() => setActiveTab("activity")}
+                        />
+                        <TabButton
+                            label="Details"
+                            isActive={activeTab === "details"}
+                            onPress={() => setActiveTab("details")}
+                        />
```

Comment:
Define tab metadata instead
 
```
const WALLET_TABS = [
  { label: "Activity", value: "activity" },
  { label: "Details", value: "details" },
] as const;

type WalletTab = typeof WALLET_TABS[number]["value"];
```


Why?
It eliminates:

Remove Magic strings → setActiveTab("activty")
Runtime bugs → full type safety with WalletTab union
Duplication → one source of truth for labels, values, future icons/badges